internal/internal_workflow.go

```diff
@@ -1742,3 +1754,56 @@ func (us updateSchedulerImpl) Spawn(ctx Context, name string, highPriority bool,
 func (us updateSchedulerImpl) Yield(ctx Context, reason string) {
 	getState(ctx).yield(reason)
 }
+
+func (m *mutexImpl) Lock(ctx Context) error {
+	err := Await(ctx, func() bool {
+		return !m.locked
+	})
+	if err != nil {
+		return err
+	}
+	m.locked = true
+	return nil
+}
+
+func (m *mutexImpl) TryLock(ctx Context) bool {
+	if m.locked {
+		return false
+	}
+	m.locked = true
+	return true
+}
+
+func (m *mutexImpl) Unlock() {
+	if !m.locked {
+		panic("Mutex.Unlock() was called on an unlocked mutex")
+	}
+	m.locked = false
+}
+
+func (m *mutexImpl) IsLocked() bool {
```

Comment:
Would be ok if you didn't expose this, but up to you

Reply:
I exposed it because I wanted a condition a user could check in an `Await` or update validator